src/typeEvaluator/typeHelpers.ts

```diff
@@ -110,6 +112,34 @@ export function mapConcrete(
   mapper: (node: ConcreteTypeNode) => TypeNode,
   mergeUnions: (nodes: TypeNode[]) => TypeNode = (nodes) =>
     optimizeUnions({type: 'union', of: nodes}),
+): TypeNode {
+  return mapUnion(
```

Comment:
Actually, how about we take the `mapUnion` here and just call it `mapConcrete`? So `mapConcrete` will call the function with either a concrete value _or_ `unknown`. And then it's up to the caller to decide how to deal with `unknown`. The more I think I about it, the less it seems like "`unknown` input implies `unknown` output" is a good assumption. In many scenarios we know way more about the output type (e.g. `round()` is either way `null` or `number` even if the input is `unknown`.